Frameworks/UIKit/StarboardXaml/ApplicationMain.mm

```diff
@@ -56,14 +61,29 @@ int ApplicationMainStart(const char* principalName,
 
     // Populate Objective C equivalent of activation argument
     if (activationType == ActivationTypeToast) {
-        NSString* toastArgument = Strings::WideToNSString(static_cast<HSTRING>(activationArg));
-        activationArgument = toastArgument;
+        WAAToastNotificationActivatedEventArgs* toastArgument = [WAAToastNotificationActivatedEventArgs createWith:activationArg];
+        NSMutableDictionary* userInput = [NSMutableDictionary new];
+        for (NSString* key in [toastArgument.userInput allKeys]) {
```

Comment:
The code that converts the IPropertyValues to NSDictionary is same, you should make a utility function that is called in both.  

---
In reply to: [72338685](https://github.com/Microsoft/WinObjC/pull/637#discussion_r72338685) [](ancestors = 72338685,72317600)